### ISSUE TITLE
How to convert a type in F# to CBORobject?

### ISSUE THREAD

**Issue by l1x** [peteroupc/peteroupc.github.io#7]
I was wondering if there is a way to convert a simple type in F# to a CBORobject.

type HitType = {
  hitHostname         : string;
  hitMemoryUsed   : float;
  hitPage                   : string;
}

Let me know if this is supported by this project. Thanks!

**Comment by peteroupc** [comment]
Did you try `CBORObject.FromObject`?

**Comment by l1x** [comment]
I did.

```
let exampleHit = { hitHostname = "google.com"; hitMemoryUsed = 0.1234; hitPage = "index.html"}

let cborTest() =
  Console.WriteLine("{0}", "Cbor test")
  let cbor = CBORObject.FromObject(exampleHit)
  let encoded = cbor.EncodeToBytes()
  let buffer = new MemoryStream()
  buffer.Write(encoded, 0, encoded.Length)
  buffer.Seek(int64(0), SeekOrigin.Begin) |> ignore
  let decoded = CBORObject.Read(buffer);
  Console.WriteLine("{0}", decoded)
```

Output:

```
Cbor test
{}
```

Maybe I am missing something.

**Comment by peteroupc** [comment]
This is currently not supported. 

`HitType` has only read-only properties, and `CBORObject.FromObject` supports read-only properties of a type only if that type is compiler-generated (e.g., in the case of anonymous types).  I will see what can be done to support F# types in the next version.

**Comment by peteroupc** [comment]
Released version 4.1.0 which should address this issue.
